lib/shelley/src/Cardano/Wallet/Shelley/Pools.hs

```diff
@@ -577,26 +605,40 @@ monitorMetadata tr gp fetchMetadata DBLayer{..} = forever $ do
         f = unActiveSlotCoefficient (getActiveSlotCoefficient gp)
 
 data StakePoolLog
-    = MsgFollow FollowLog
+    = MsgDb PoolDbLog
+    | MsgFollow FollowLog
     | MsgStartMonitoring [BlockHeader]
     | MsgHaltMonitoring
     | MsgCrashMonitoring
     | MsgRollingBackTo SlotNo
+    | MsgStakePoolGarbageCollection PoolGarbageCollectionInfo BracketLog
     | MsgStakePoolRegistration PoolRegistrationCertificate
     | MsgStakePoolRetirement PoolRetirementCertificate
     | MsgErrProduction ErrPointAlreadyExists
     | MsgFetchPoolMetadata StakePoolMetadataFetchLog
     | MsgFetchTakeBreak Int
     deriving (Show, Eq)
 
+data PoolGarbageCollectionInfo = PoolGarbageCollectionInfo
+    { currentEpoch :: EpochNo
+        -- ^ The current epoch at the point in time the garbage collector
+        -- was invoked.
+    , removalEpoch :: EpochNo
```

Comment:
**Suggestion:** The name `removalEpoch` seems a little unclear. Perhaps call it `latestRetirementEpoch`?

**Justification:** the name `latestRetirementEpoch` implies a relationship with `retirementEpoch` in `PoolRetirementCertificate`.

```suggestion
    , latestRetirementEpoch :: EpochNo
```

Reply:
> **Suggestion:** The name `removalEpoch` seems a little unclear. Perhaps call it `latestRetirementEpoch`?

Fixed in 78e8e04f888c7288166147b63d22fdff6ccdc7c2.